### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by DaniPopes**
I ran [`typos`](https://github.com/crate-ci/typos) to check for typos in this repo.

The first 3 commits are mostly automatic changes from running `typos -w ./<dir>`, while the rest are fixes that I've manually applied.

I have not included the config file because it's pretty ugly with a lot of manual exclusions but I can post it here if needed

### CODE REVIEW

File: library/core/src/primitive_docs.rs [hunk 1]
```diff
@@ -1428,8 +1428,8 @@ mod prim_ref {}
 /// let ptr: fn(usize) -> usize = add_one;
 /// assert_eq!(ptr(5), 6);
 ///
-/// let clos: fn(usize) -> usize = |x| x + 5;
```
Comment (compiler-errors) [inline library/core/src/primitive_docs.rs:1431]: there's no reason to rename this random variable imo

File: library/core/src/str/mod.rs [hunk 2]
```diff
@@ -1146,7 +1146,7 @@ impl str {
     ///
     /// assert_eq!(s.find('L'), Some(0));
     /// assert_eq!(s.find('é'), Some(14));
-    /// assert_eq!(s.find("pard"), Some(17));
+    /// assert_eq!(s.find("part"), Some(17));
```
Comment (compiler-errors) [inline library/core/src/str/mod.rs:1149]: what leads you to believe that this should've been named "part"? this is not right.

File: compiler/rustc_target/src/spec/abi.rs [hunk 3]
```diff
@@ -300,21 +296,21 @@ impl Abi {
...
-            "Abi index did not match `AbiDatas` ordering"
+            "Abi index did not match `AbiData` ordering"
         );
         i
     }
 
     #[inline]
-    pub fn data(self) -> &'static AbiData {
-        &AbiDatas[self.index()]
+    pub fn data(self) -> &'static AbiDatum {
```
Comment (compiler-errors) [inline compiler/rustc_target/src/spec/abi.rs:312]: The scope of this PR is way too wide. Renaming structs in the compiler shouldn't be mixed with fixing comments.

File: compiler/rustc_target/src/spec/abi.rs [hunk 4]
```diff
@@ -63,67 +63,63 @@ impl Abi {
...
-    AbiData { abi: Abi::CCmseNonSecureCall, name: "C-cmse-nonsecure-call" },
-    AbiData { abi: Abi::Wasm, name: "wasm" },
-    AbiData { abi: Abi::System { unwind: false }, name: "system" },
-    AbiData { abi: Abi::System { unwind: true }, name: "system-unwind" },
-    AbiData { abi: Abi::RustIntrinsic, name: "rust-intrinsic" },
-    AbiData { abi: Abi::RustCall, name: "rust-call" },
-    AbiData { abi: Abi::PlatformIntrinsic, name: "platform-intrinsic" },
-    AbiData { abi: Abi::Unadjusted, name: "unadjusted" },
-    AbiData { abi: Abi::RustCold, name: "rust-cold" },
+const AbiData: &[AbiDatum] = &[
```
Comment (Nilstrieb) [inline compiler/rustc_target/src/spec/abi.rs:74]: This is not correct, AbiData is completely fine and correct.

**Review by Noratrieb — CHANGES_REQUESTED** [review]
Thank you for the PR.
This PR touches quite some things and is quite big. Could you split it up into separate PRs for compiler/library/rustdoc? Additionally, I see changes that seem incorrect, I've highlighted one of them. Could you check all the changes yourself again and make sure that you don't see any other problems? I've slimmed through some of the diff and most changes look good though.

File: compiler/rustc_target/src/spec/abi.rs (hunk 4 above)
Comment (DaniPopes) [reply compiler/rustc_target/src/spec/abi.rs:74]: `AbiData` (plural) is a list of `AbiDatum` (singular), "datas" doesn't exist. Should I still revert it?

File: compiler/rustc_target/src/spec/abi.rs (hunk 3 above)
Comment (DaniPopes) [reply compiler/rustc_target/src/spec/abi.rs:312]: I'll split it into separate PRs for compiler/library/rustdoc as @Nilstrieb mentioned

File: compiler/rustc_target/src/spec/abi.rs (hunk 4 above)
Comment (compiler-errors) [reply compiler/rustc_target/src/spec/abi.rs:74]: Yes

**Pull request closed without merging**